src/mint.rs

```diff
@@ -0,0 +1,190 @@
+use codec::{Decode, Encode};
+use runtime_primitives::traits::{SimpleArithmetic, Zero};
+use srml_support::ensure;
+
+#[derive(Encode, Decode, Copy, Clone, Debug, Eq, PartialEq)]
+pub enum AdjustCapacityBy<Balance> {
+    // Set capacity of mint to specific value
+    Setting(Balance),
+    // Add to the capacity of the mint
+    Adding(Balance),
+    // Reduce capacity of the mint
+    Reducing(Balance),
+}
+
+#[derive(Encode, Decode, Copy, Clone, Debug, Eq, PartialEq)]
+pub struct AdjustOnInterval<Balance, BlockNumber> {
+    pub block_interval: BlockNumber,
+    pub adjustment_type: AdjustCapacityBy<Balance>,
+}
+
+#[derive(Encode, Decode, Copy, Clone, Debug, Eq, PartialEq)]
+pub enum Adjustment<Balance, BlockNumber> {
+    // First adjustment will be after AdjustOnInterval.block_interval
+    Interval(AdjustOnInterval<Balance, BlockNumber>),
+    // First Adjustment will be at absolute blocknumber
+    IntervalAfterFirstAdjustmentAbsolute(AdjustOnInterval<Balance, BlockNumber>, BlockNumber),
+    // First Adjustment will be after a specified number of blocks
+    IntervalAfterFirstAdjustmentRelative(AdjustOnInterval<Balance, BlockNumber>, BlockNumber),
+}
+
+#[derive(Encode, Decode, Default)]
+// Note we don't use TokenMint<T: Trait> it breaks the Default derivation macro with error T doesn't impl Default
+// Which requires manually implementing Default trait.
+// We want Default trait on TokenMint so we can use it as value in StorageMap without needing to wrap it in an Option
+pub struct Mint<Balance, BlockNumber>
+where
+    Balance: Copy + SimpleArithmetic,
+    BlockNumber: Copy + SimpleArithmetic,
+{
+    capacity: Balance,
+
+    adjustment_on_interval: Option<AdjustOnInterval<Balance, BlockNumber>>,
```

Comment:
I wanted to allow for creating a mint that didn't adjust on intervals, (of course that can be done by make adjust_capacity_at_block_number = None)

You are correct about the inconsistent state, that is why I introduced the Adjustment enum as a way to make sure client cannot create that state. But I see now its a kind of band-aid it will be better to make it impossible to actually have that state.

Reply:
Actually the `Adjustment` type was for another purpose..

Reply:
..but we will loose option to have a state where the mint can be "paused" without modifying the adjustment policy.

Reply:
done.